### PR TITLE
Dataset fixes

### PR DESCRIPTION
**Pull request by ngupta23**
Fixes: #1763 

#### Describe the changes you've made
- Restructured `get_dataset` to be more understandable
- Created a unified flow (e.g. removed `index` argument and merged with existing functionality.
- Updated documentation and unit tests

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Unit tests have been updated and tested locally for datasets

## Checklist:
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by ngupta23** [comment]
@Yard1 The test passes locally but fails no GitHub. Would you know why? Thanks!

**Comment by ngupta23** [comment]
@Yard1 Looks like the local file read check (new addition) is failing on GitHub (passing locally). I have disabled it for now. If you think we can fix it, please do it, else we can merge it as is for now. Thanks!

File: pycaret/datasets.py [hunk 1]
```diff
@@ -57,31 +78,41 @@ def get_data(
...
 
     extension = ".csv"
+    filename = str(dataset) + extension
 
-    if index:
-        filename = "index.csv"
+    folder = "common" if folder is None else folder
+
+    if dataset == "index":
         complete_address = root + meta_dir + folder + "/" + filename
```
Comment (Yard1) [inline pycaret/datasets.py:103]: use os.path.join?

**Review by Yard1 — APPROVED** [review]
Looks great, thanks!

File: pycaret/datasets.py (hunk 1 above)
Comment (ngupta23) [reply pycaret/datasets.py:103]: It is a URL. Sos os.path.join does not work. For example, I get this 
`https://raw.githubusercontent.com/pycaret/datasets/main/data\time_series/seasonal\1.csv` as the URL and it fails to retrieve it.

**Pull request merged**